pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -870,7 +870,7 @@ private boolean isPrimary(QueryExecutor queryExecutor) throws SQLException, IOEx
     if (inHotStandby != null && inHotStandby.equalsIgnoreCase("on")) {
       return false;
     }
-    // Host may not be in recovery but still have transaction_read_only = 'off' set,
+    // Host may not be in recovery but still have 'transaction_read_only=on' set,
```

Comment:
I'm wondering if this isn't the other way around
Host may be in recovery but still have transaction_read_only=off ?

Is it possible to find the reference for this ?

Reply:
https://www.postgresql.org/docs/current/hot-standby.html suggests that the reason you need to check this is because it can change, not sure about the "host may not be in recovery" bit ?

Reply:
>  Host may be in recovery but still have transaction_read_only=off ?

Docs says this is impossible
> During hot standby, the parameter transaction_read_only is always true and may not be changed.

Probably I failed to explain the logic in this comments: 
As per docs: If `in_hot_standby=on`, then `transaction_read_only` is always `on` or "replica is always readonly". Hence we shortcut once `in_hot_standby=on` has been observed in parameter statuses.
Primary server may be in readonly mode ie `in_hot_standby=off` and `transaction_read_only=on`. Could be set temporary during backup restore on some other manual database administering. This is what I checked in added test case. So I guess `in_hot_standby` was added because primary could be in a readonly mode and checking `transaction_read_only` is not sufficient in such cases:
> Users can determine whether hot standby is currently active for their session by issuing SHOW in_hot_standby. (In server versions before 14, the in_hot_standby parameter did not exist; a workable substitute method for older servers is SHOW transaction_read_only.)

I would be happy if we could rephrase that comment in terse and clear form.